src/utl/include/utl/histogram.h

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2023-2025, The OpenROAD Authors
+
+#pragma once
+
+#include <algorithm>
+#include <optional>
+#include <string>
+#include <vector>
+
+namespace utl {
+
+class Logger;
+
+template <typename DataType>
+class Histogram
+{
```

Comment:
As you are using a cpp with explicit instantiation it would be good to comment here that this can only be used with DataType = int | float.